testsuite/src/test/java/org/jboss/intersmash/testsuite/provision/openshift/KeycloakRealmImportOperatorProvisionerTest.java

```diff
@@ -0,0 +1,373 @@
+ * limitations under the License.
+ */
+package org.jboss.intersmash.testsuite.provision.openshift;
+
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Objects;
+
+import org.jboss.intersmash.tools.application.openshift.KeycloakRealmImportOperatorApplication;
+import org.jboss.intersmash.tools.application.openshift.PostgreSQLImageOpenShiftApplication;
+import org.jboss.intersmash.tools.junit5.IntersmashExtension;
+import org.jboss.intersmash.tools.provision.openshift.KeycloakRealmImportOperatorProvisioner;
+import org.jboss.intersmash.tools.provision.openshift.PostgreSQLImageOpenShiftProvisioner;
+import org.jboss.intersmash.tools.provision.openshift.operator.resources.OperatorGroup;
+import org.jboss.intersmash.tools.util.tls.CertificatesUtils;
+import org.junit.jupiter.api.AfterAll;
+import org.junit.jupiter.api.AfterEach;
+import org.junit.jupiter.api.Assertions;
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.keycloak.k8s.v2alpha1.Keycloak;
+import org.keycloak.k8s.v2alpha1.KeycloakRealmImport;
+import org.keycloak.k8s.v2alpha1.KeycloakRealmImportSpec;
+import org.keycloak.k8s.v2alpha1.KeycloakSpec;
+import org.keycloak.k8s.v2alpha1.keycloakrealmimportspec.Realm;
+import org.keycloak.k8s.v2alpha1.keycloakrealmimportspec.realm.Users;
+import org.keycloak.k8s.v2alpha1.keycloakrealmimportspec.realm.users.Credentials;
+import org.keycloak.k8s.v2alpha1.keycloakspec.Db;
+import org.keycloak.k8s.v2alpha1.keycloakspec.Hostname;
+import org.keycloak.k8s.v2alpha1.keycloakspec.Http;
+import org.keycloak.k8s.v2alpha1.keycloakspec.Ingress;
+import org.keycloak.k8s.v2alpha1.keycloakspec.db.PasswordSecret;
+import org.keycloak.k8s.v2alpha1.keycloakspec.db.UsernameSecret;
+import org.slf4j.event.Level;
+
+import cz.xtf.core.openshift.OpenShiftWaiters;
+import cz.xtf.core.openshift.OpenShifts;
+import cz.xtf.core.waiting.SimpleWaiter;
+import cz.xtf.junit5.annotations.CleanBeforeAll;
+import io.fabric8.kubernetes.api.model.DeletionPropagation;
+import io.fabric8.kubernetes.api.model.KubernetesResourceList;
+import io.fabric8.kubernetes.client.dsl.NonNamespaceOperation;
+import io.fabric8.kubernetes.client.dsl.Resource;
+import lombok.extern.slf4j.Slf4j;
+
+/**
+ * Test the Keycloak Operator provisioning model and APIs
+ *
+ * See <br>
+ *     - https://github.com/keycloak/keycloak-operator/tree/master/deploy/examples
+ * <p>
+ * Not all examples are actually run, but these tests are more about the framework functionality verification than
+ * about the Keycloak operator testing.
+ */
+@Slf4j
+@CleanBeforeAll
+//@Disabled("WIP - Disabled until global-test.properties is configured with the required property")
+public class KeycloakRealmImportOperatorProvisionerTest {
+	private static KeycloakRealmImportOperatorProvisioner KEYCLOAK_OPERATOR_PROVISIONER;
```

Comment:
You can remove any parameters from `initializeOperatorProvisioner` below and initialize this directly with a dummy configured application descriptor, see the [InifinispanOperatorTest](https://github.com/Intersmash/intersmash/blob/main/testsuite/src/test/java/org/jboss/intersmash/testsuite/provision/openshift/InfinispanOperatorProvisionerTest.java#L66) for inspiration.

Reply:
I intentionally removed the dummy and initialize the `KeycloakRealmImportOperatorProvisioner` with the correct resource expected in the specific test;

Actually I found that dummy initialization quite confusing;

Reply:
It can be confusing, I agree. But you could just `mock` it, for instance. 

Furthermore, we should consider that this kind of tests (Operator based provisioner tests) should only test operations involving the CRs, while a lifecycle test should be implemented for all provisioners in https://github.com/tommaso-borgato/intersmash/blob/issues-37/testsuite/src/test/java/org/jboss/intersmash/testsuite/provision/openshift/ProvisionerCleanupTestCase.java

I am aware this is even more confusing and this is why I logged https://github.com/Intersmash/intersmash/issues/47

Basically, really the reference implementation here should be the one in [InifinispanOperatorTest](https://github.com/Intersmash/intersmash/blob/main/testsuite/src/test/java/org/jboss/intersmash/testsuite/provision/openshift/InfinispanOperatorProvisionerTest.java#L66), i.e. in terms of lifecycle: `configure` in `@BeforeAll`, `dismiss` in `@AfterAll`and then - in each test method - we should have:  `subscribe -> create and verify CRs -> unsubscribe`. That is to say no `deploy`or `undeploy` should be called at all. 

Can you please try and fix the test this way? We could then use this implementation as the foundation for the work needed by #47

Reply:
After reviewing the code more in detail, the issue is that here we do have two provisioners, one for Keycloak and one for PostgreSql. The latter is only used  by the second test method, therefore its life cycle (_configure -> subscribe -> preDeploy -> deploy -> undeploy -> postUndeploy -> unsubscribe -> dismiss_) can be fully handled inside such method.

Regarding the Keycloak provisioner, given the initialization is done at the test method level, here as well is where the life cycle (_configure -> subscribe -> create/verify CRs -> unsubscribe -> dismiss_) will be handled, while all the related instructions will be removed from `@BeforeAll` and `@AfterAll` methods.

Reply:
Thanks, the latest push fix this, resolving.